Add locator deprecation warning and additional examples

anotherkowski opened the pull request:
Selenium 4.0.0 deprecates `find_element_by_*` methods. This PR makes a handful of edits to address that:
- Add notice to introduction about deprecated methods
- Add table of locator methods to page introduction
- Add additional examples with preferred methods under each locator example
- Add additional XPath browser extensions
- Move and convert inline notes into comments directly beneath relevant code block
- Add relevant alerts/notices as footnotes to improve readability

anotherkowski commented:
Note: I did **not** remove the legacy `find_element_by_*` locator methods because they do still behave as expected for earlier versions of Selenium, though if used they will print a warning to the console:

` DeprecationWarning: find_element_by_* commands are deprecated. Please use find_element() instead.`

anotherkowski commented:
@baijum @anarang - PR for consideration at your convenience

anarang commented:
@anotherkowski thanks for raising the PR. The changes about the deprecation notice and the table look good to me at the top and once at the bottom of the PR. However, I am not sure of the cosmetic changes to the text that you propose, changing '' to ``, there is some inconsistency there.

baijum commented:
@anotherkowski  Thanks for the PR! Now that I have merged a similar PR #99 So, can you please rebase your PR branch?

Also, the reStructuredText changes should be a separate PR.

Thanks for @anarang for the review!

anotherkowski commented:
> @anotherkowski thanks for raising the PR. The changes about the deprecation notice and the table look good to me at the top and once at the bottom of the PR. However, I am not sure of the cosmetic changes to the text that you propose, changing '' to ``, there is some inconsistency there.

Absolutely - understood. I will revert them for consistency's sake with the rest of the docs.

anotherkowski commented:
Rebased, resolved conflicts, and reverted the rogue `` into " at @anarang's suggestion. I maintained the table at the top of the Locating Elements page, but also preserved the updates made by @dysphere. 

Please let me know if there are other things I can do to shore up this PR. I appreciate your consideration!

anarang commented on source/locating-elements.rst:
```diff
@@ -215,11 +240,15 @@ For instance, consider this page source::
   continue_link = driver.find_element(By.LINK_TEXT, 'Continue')
   continue_link = driver.find_element(By.PARTIAL_LINK_TEXT, 'Conti')
 
+  continue_link = driver.find_element(By.LINK_TEXT, 'Continue')
```
isn't this repetition of line 245, 246.

anarang commented on source/locating-elements.rst:
```diff
@@ -237,10 +266,13 @@ For instance, consider this page source::
   </html>
 
-The heading (h1) element can be located like this::
+The heading "h1" element can be located like this::
```
what is the need for changing (h1) to "h1"?

anarang commented on source/locating-elements.rst:
```diff
@@ -237,10 +266,13 @@ For instance, consider this page source::
 
   heading1 = driver.find_element(By.TAG_NAME, 'h1')
 
+  heading1 = driver.find_element(By.TAG_NAME, 'h1')
```
I see a repetition again, line 271 repeated.

anarang replied:
I see a few repetitions below as well, please revisit the PR thoroughly.

baijum commented:
I closing this PR for now. Please reopen once the review comments are addressed.
